wpilibc/src/main/native/include/frc/util/Color.h

```diff
@@ -761,30 +771,32 @@ class Color {
    * @return The color
    */
   static constexpr Color FromHSV(int h, int s, int v) {
-    if (s == 0) {
-      return {v / 255.0, v / 255.0, v / 255.0};
-    }
+    // loosly based on
+    // https://en.wikipedia.org/wiki/HSL_and_HSV#HSV_to_RGB
 
+    int chroma = (s * v) >> 8;
     int region = h / 30;
-    int remainder = (h - (region * 30)) * 6;
-
-    int p = (v * (255 - s)) >> 8;
-    int q = (v * (255 - ((s * remainder) >> 8))) >> 8;
-    int t = (v * (255 - ((s * (255 - remainder)) >> 8))) >> 8;
+    // remainder converted from 0-30 to roughly 0-255
+    int remainder = (h - (region * 30)) * 8;
+    // lowest value of r, g or b
+    int m = v - chroma;
+    // part in each region that changes
+    // goes from 0 to chroma
```

Comment:
What does this mean?

Reply:
in each 60° hue region there is one rgb colour component that is m, one that is v and one that changes. X is the part that changes ![image](https://user-images.githubusercontent.com/34265013/192910053-60f06806-81bf-4bd0-950c-1b9679e3e64b.png)

Reply:
That should be documented in the code.

Reply:
Are the comments more clear now?